### PR TITLE
Machine timer

### PR DESCRIPTION
**Pull request by IFX-Anusha**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Timer Module Fix with tests

### CODE REVIEW

File: tests/psoc6/timer.py [hunk 1]
```diff
@@ -1,27 +1,54 @@
...
 
-t = Timer(0)
-t.init(period=2000, mode=Timer.ONE_SHOT, callback=lambda t: print("Oneshot Timer"))
-time.sleep(30)
-t.deinit()
 
+def test_periodic():
+    # Periodic timer
+    global periodic_triggered
+    tim_periodic = Timer(0, period=1, mode=Timer.PERIODIC, callback=call_periodic)
```
[the remark this answers is not in the document]
Comment (IFX-Anusha) [reply tests/psoc6/timer.py:37]: Yes We can set any values

File: ports/psoc6/machine_timer.c [hunk 2]
```diff
@@ -40,27 +40,45 @@ static mp_obj_t machine_timer_init_helper(machine_timer_obj_t *self, size_t n_ar
...
+        self->period = args[ARG_period].u_int;
+        period = (float)args[ARG_period].u_int / 1000.0f;
+    }
 
     if (args[ARG_callback].u_obj != mp_const_none) {
         self->callback = args[ARG_callback].u_obj;
     }
 
+
+    uint32_t period_hal;
```
[the remark this answers is not in the document]
Comment (IFX-Anusha) [reply ports/psoc6/machine_timer.c]: Yes Updated

[the remark this answers is not in the document]
Comment (jaenrig-ifx) [reply ports/psoc6/machine_timer.c]: Perfect! I will fully approve, when i reconnect everything to the hil :)

**Review by jaenrig-ifx — APPROVED** [review]
Merging the timer, as the errors come from  pwm tests, which is addressed in #126

**Pull request merged**